File: codes/golden/java/HumanEval_2_truncate_number.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static float truncateNumber(float number) {
        return number % 1.0f;
    }
    public static void main(String[] args) {
    assert(truncateNumber((3.5f)) == (0.5f));
    assert(truncateNumber((1.25f)) == (0.25f));
    assert(truncateNumber((123.0f)) == (0.0f));
    }

}

File: codes/golden/cpp/HumanEval_2_truncate_number.cpp
#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
float truncate_number(float number) {
    return number - floor(number);
}
int main() {
    auto candidate = truncate_number;
    assert(candidate((3.5)) == (0.5));
    assert(candidate((1.25)) == (0.25));
    assert(candidate((123.0)) == (0.0));
}
